src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/WindowsFormsApplicationBase.vb

```
@@ -829,6 +872,160 @@ Namespace Microsoft.VisualBasic.ApplicationServices
+            'var sidAuthUsers = new SecurityIdentifier(WellKnownSidType.AuthenticatedUserSid, null);
+            Dim sidNetworkService As New SecurityIdentifier(WellKnownSidType.NetworkServiceSid, Nothing)
+            Dim sidWorld As New SecurityIdentifier(WellKnownSidType.WorldSid, Nothing)
+
+            Dim pipeSecurity As New PipeSecurity
+
+            ' Deny network access to the pipe
+            Dim accessRule As New PipeAccessRule(sidNetworkService, PipeAccessRights.ReadWrite, AccessControlType.Deny)
+            pipeSecurity.AddAccessRule(accessRule)
+
+            ' Allow Everyone to read/write
+            accessRule = New PipeAccessRule(sidWorld, PipeAccessRights.ReadWrite, AccessControlType.Allow)
+            pipeSecurity.AddAccessRule(accessRule)
+
+            ' Current user is the owner
+            Dim sidOwner As SecurityIdentifier = WindowsIdentity.GetCurrent().Owner
+            If sidOwner IsNot Nothing Then
+                accessRule = New PipeAccessRule(sidOwner, PipeAccessRights.FullControl, AccessControlType.Allow)
+                pipeSecurity.AddAccessRule(accessRule)
+            End If
+
+            Try
+                ' Create pipe and start the async connection wait
+                _namedPipeServerStream = New NamedPipeServerStream(
+                        pipeName:=_namedPipeID,
+                        direction:=PipeDirection.In,
+                        maxNumberOfServerInstances:=1,
+                        transmissionMode:=PipeTransmissionMode.Byte,
+                        options:=PipeOptions.Asynchronous,
+                        inBufferSize:=0,
+                        outBufferSize:=0)
+                ' We are the first instance with the named pipe server listening and allow the form to load
+                ' Begin async wait for connections
+                _namedPipeServerStream.BeginWaitForConnection(AddressOf namedPipeServerConnectionCallback, _namedPipeServerStream)
+            Catch ex As Exception
```

>Exception [](start = 24, length = 9)

Please catch specific exceptions that are expected. It looks like that is only `IOException` in the case of a race condition when two instances of the application start at the same time. In that case, the caller should connect to the instance that was successful instead of failing silently.

---

Since there is a Mutex here, there is no race to on the creation, it is more about the application exiting but I did change Catch.